works/views.py

```diff
@@ -99,6 +103,69 @@ def update(self, request, pk=None):
     def partial_update(self, request, pk=None):
         return self.update(request, pk)
 
+    @detail_route(methods=['get'], url_path='report')
+    def report(self, request, pk=None):
```

Comment:
Could this function be improved? maybe split writer functions into smaller functions? I dunno